### ISSUE TITLE
Nullable warning reported in Error List with "Build+Intellisense" filter

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#31765]
Reported by @arivoir in https://github.com/dotnet/docs/issues/9517

As I understood the repro steps:
- using dev16 preview1 or preview1.1
- clone https://github.com/arivoir/Open.Flickr
- build (notice there are no nullable warnings in Output)
- open the Error List:
  - notice there are no nullable warnings when filtered with "Build"
  - but there are unexpected nullable warnings when filtered with "Build+Intellisense"

@arivoir Please confirm this is an accurate summary. It would be great if you could attach a small screenshot of the error list.
Also, did you try repairing your instance of preview1.1?

I was not able to repro on preview1 or preview2. So I'm looking for troubleshooting steps to provide @arivoir. 
@heejaechang is there a way to log some information from the build that powers the "Intellisense" list?

**Comment by arivoir** [comment]
@jcouv Here there is a capture of my "Error List"
![image](https://user-images.githubusercontent.com/17496741/49966654-60443b00-feff-11e8-86ee-6240295b2fef.png)
I didn't repair my instance yet. Can you tell me exactly what should I do?

**Comment by jcouv** [comment]
Just to confirm, those warnings also appear if you filter on "IntelliSense Only". Correct?

For repairing, you launch the VS installer (hit Windows key, "installer", it should come up. Then it is straightforward: there's a repair option under "More" for preview1.1 instance.

**Comment by arivoir** [comment]
Yes, they also appear when filter by "IntelliSense Only".

**Comment by Vannevelj** [comment]
I'm seeing the same issue although sometimes the errors disappear. If you open the solution you see errors but if you then make a change to a file (can be unrelated to the `?`) it will remove the warnings. From that point forward, any file you open that has these errors will have them disappear. This lasts until you do another build: you need to make an edit to a file before it will remove the warnings again.

![image](https://user-images.githubusercontent.com/2777107/51427462-36fa8700-1bf0-11e9-9c0c-cf2a75952495.png)

This is the csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <VersionPrefix>3.0.0</VersionPrefix>
    <FileVersion>3.0.0</FileVersion>
    <!-- The AssemblyVersion should only be updated on a Major version bump -->
    <AssemblyVersion>3.0.0.0</AssemblyVersion>
    <TargetFramework>netstandard1.6</TargetFramework>
    <LangVersion>8.0</LangVersion>
    <NullableReferenceTypes>True</NullableReferenceTypes>
    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
    <DebugType>full</DebugType>
    <AssemblyName>MyAssembly.Client</AssemblyName>
    <PackageId>MyAssembly.Client</PackageId>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="OtherPackage" Version="[1.10.0, 2]" />
    <PackageReference Include="OtherPackage2" Version="2.6.3" />
    <PackageReference Include="OtherPackage3" Version="1.0.0" />
  </ItemGroup>
  <ItemGroup>
    <DotNetCliToolReference Include="OtherPackage4" Version="4.5.0" />
  </ItemGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <DebugSymbols>true</DebugSymbols>
    <DebugType>portable</DebugType>
  </PropertyGroup>

</Project>
```

Running in VisualStudio.16.Preview/16.0.0-pre.1.1+28408.50.

I can confirm that the errors originate from `Build` and not `Intellisense`. When I do a rebuild I can see that all projects in the solution built successfully.

**Comment by jcouv** [comment]
@heejaechang Would you be able to take a look at this issue and advise? Thanks

**Comment by emre1702** [comment]
I'm getting this problem after updating Visual Studio to preview 16.2 (from the previous latest version).
Everything worked for me before the update, but now I'm getting the same warning/error (CS8632) in Build(!) + Intellisense.

My .csproj contains:
```
 <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
    <AssemblyName>MyProject</AssemblyName>
    <RootNamespace>MyProject</RootNamespace>
    <Platforms>AnyCPU;x64</Platforms>
    <LangVersion>preview</LangVersion>
    <NullableContextOptions>enable</NullableContextOptions>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>
```
Tried it with LangVersion 8.0 and NullableReferenceTypes instead, too - same behaviour.

**Comment by KexyBiscuit** [comment]
@emre1702 Me too, more reports here: [NullableContextOptions no longer works](https://developercommunity.visualstudio.com/content/problem/576567/nullablecontextoptions-no-longer-works.html)

**Comment by moh-hassan** [comment]
The warning appear  in vs 2019 RTM v16.0.2
>Warning	CS8632	The annotation for nullable reference types should only be used in code within a '#nullable' context.	
appear in both output pan and ErrorList pan (warning)

When click on the link CS8632	, it show [google search page](http://www.bing.com/search?q=CS8632+C%23+The+annotation+for+nullable+reference+types+should+only+be+used+in+code+within+a+context.&form=VSHELP)

**Comment by jcouv** [comment]
@arivoir Are you still seeing this behavior with the Error List? I've still not been able to repro this issue. Consider trying on a newer version/preview of VS.

**Comment by JohnGalt1717** [comment]
I'm not seeing this with 16.2 preview 2.

**Comment by arivoir** [comment]
@jcouv This looks fixed in VS 16.2 Preview 2.0. Thanks

**Comment by jcouv** [comment]
Awesome. I'll go ahead and close. Thanks

**Comment by jnegron357** [comment]
I'm having this issue with 16.4.2. I modified the csproj file as needed, performed a build and still have the message. Closed VS and re-opened it, made a new build and still see the message and squiggly under the question mark. Please advise.

**Comment by jcouv** [comment]
Stumbled on this issue after a while, and now think this may have been related to https://github.com/dotnet/project-system/issues/3680 (where the IDE doesn't give the compiler the same options in the same order as the build process does).
